src/QsCompiler/DataStructures/DataTypes.fs

```diff
@@ -53,61 +53,118 @@ type QsNullable<'T> = // to avoid having to include the F# core in the C# part o
         | Some v -> Value v
         | None -> Null
 
+/// Operations for QsNullable<'T>.
+module QsNullable =
+    /// Applies 'f' to both nullable arguments if both have a Value. Returns Null otherwise.
+    let Map2 f a b =
```

Comment:
nit: It may be useful to document the typing here, noting that Map2 returns a `QsNullable<'U>` for all `f : 'T * 'T -> 'U`.

Reply:
The type is more general than that (`a` and `b` don't have to be the same type, so `f : 'a -> 'b -> 'c`), but it's rare to mention types in F# if they are inferred correctly. Do you mean as a type annotation or part of the doc comment?

Reply:
I meant as part of the doc comment, but if it is abnormal to talk about types in F# then maybe it is just that I need to adjust my way of thinking. I am used to understanding code in terms of types, so when I approached this I was thinking, "What are the types?". But maybe that isn't the F# way?

Reply:
I think in terms of types too, and in Haskell this function would have an explicit type annotation, but the F# way seems to be to omit type information whenever possible. I would prefer not to include types in the doc comment because then the types mentioned would not be checked by the compiler and could be wrong/out of date, whereas type annotations are checked for correctness by the compiler.